account_mass_reconcile/models/mass_reconcile.py

```diff
@@ -0,0 +1,334 @@
+# © 2012-2016 Camptocamp SA (Guewen Baconnier, Damien Crier, Matthieu Dietrich)
+# © 2010 Sébastien Beau
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from datetime import datetime
+import logging
+
+from odoo import models, api, fields, _
+from odoo.exceptions import Warning as UserError
+from odoo import sql_db
+
+_logger = logging.getLogger(__name__)
+
+
+class MassReconcileOptions(models.AbstractModel):
+    """Options of a reconciliation profile
+
+    Columns shared by the configuration of methods
+    and by the reconciliation wizards.
+    This allows decoupling of the methods and the
+    wizards and allows to launch the wizards alone
+    """
+    _name = 'mass.reconcile.options'
+
+    @api.model
+    def _get_rec_base_date(self):
+        return [
+            ('newest', 'Most recent move line'),
+            ('actual', 'Today'),
+        ]
+
+    write_off = fields.Float(
+        'Write off allowed',
+        default=0.,
+    )
+    account_lost_id = fields.Many2one(
+        'account.account',
+        string="Account Lost",
+    )
+    account_profit_id = fields.Many2one(
+        'account.account',
+        string="Account Profit",
+    )
+    journal_id = fields.Many2one(
+        'account.journal',
+        string="Journal",
+    )
+    date_base_on = fields.Selection(
+        '_get_rec_base_date',
+        required=True,
+        string='Date of reconciliation',
+        default='newest',
+    )
+    filter = fields.Char(
+        string='Filter',
+    )
+    income_exchange_account_id = fields.Many2one(
+        'account.account',
+        string='Gain Exchange Rate Account',
+    )
+    expense_exchange_account_id = fields.Many2one(
+        'account.account',
+        string='Loss Exchange Rate Account',
+    )
+
+
+class AccountMassReconcileMethod(models.Model):
+    _name = 'account.mass.reconcile.method'
+    _description = 'reconcile method for account_mass_reconcile'
+    _inherit = 'mass.reconcile.options'
+    _order = 'sequence'
+
+    @staticmethod
+    def _get_all_rec_method():
+        return [
+            ('mass.reconcile.simple.name', 'Simple. Amount and Name'),
+            ('mass.reconcile.simple.partner', 'Simple. Amount and Partner'),
+            ('mass.reconcile.simple.reference',
+             'Simple. Amount and Reference'),
+            ('mass.reconcile.advanced.ref',
+             'Advanced. Partner and Ref.'),
+        ]
+
+    def _get_rec_method(self):
+        return self._get_all_rec_method()
```

Comment:
this indirection is not needed

Reply:
Not sure what you mean here

Reply:
you can merge "_get_all_rec_method" into this method and  rename it `_selection_name` (I don't see those methods used somewhere else)

Reply:
hmm, thinking about it twice: `_get_all_rec_method` it's kind of an api method, so:
1. rename `_get_rec_method` to `_selection_name`
2. rename `_get_all_rec_method` to `_get_reconcilation_methods`